hack/ci-e2e.sh

```diff
@@ -32,58 +43,97 @@ minikube start
 export BMC_USER=admin
 export BMC_PASSWORD=password
-CIRROS_VERSION="0.6.2"
-IMAGE_FILE="cirros-${CIRROS_VERSION}-x86_64-disk.img"
-export IMAGE_CHECKSUM="c8fc807773e5354afe61636071771906"
-export IMAGE_URL="http://${IP_ADDRESS}/${IMAGE_FILE}"
-IMAGE_FOLDER="${REPO_ROOT}/test/e2e/images"
 
-## Setup image server
-# Create a directory for images
+if [[ "${BMO_E2E_EMULATOR}" == "vbmc" ]]; then
+  # Start VBMC
+  docker run --name vbmc --network host -d \
+    -v /var/run/libvirt/libvirt-sock:/var/run/libvirt/libvirt-sock \
+    -v /var/run/libvirt/libvirt-sock-ro:/var/run/libvirt/libvirt-sock-ro \
+    quay.io/metal3-io/vbmc
+
+  VBMC_PORT="16230"
+  # Together with the VBMC_PORT this becomes the BMC_ADDRESS used by the BMH in the test.
+  export BMC_ADDRESS="ipmi://${IP_ADDRESS}:${VBMC_PORT}"
+
+  # Add BMH VM to VBMC
+  docker exec vbmc vbmc add "${VM_NAME}" --port "${VBMC_PORT}"
+  docker exec vbmc vbmc start "${VM_NAME}"
+  docker exec vbmc vbmc list
+
+elif [[ "${BMO_E2E_EMULATOR}" == "sushy-tools" ]]; then
+  # Sushy-tools variables
+  SUSHY_EMULATOR_FILE="${REPO_ROOT}"/test/e2e/sushy-tools/sushy-emulator.conf
+  SUSHY_PORT="8000"
+
+  # Start sushy-tools
+  # Privileged mode is needed to access libvirt.
+  sudo docker run --name sushy-tools -d --privileged --network host \
+    -v "${SUSHY_EMULATOR_FILE}":/etc/sushy/sushy-emulator.conf:Z \
+    -v /var/run/libvirt:/var/run/libvirt:Z \
+    -e SUSHY_EMULATOR_CONFIG=/etc/sushy/sushy-emulator.conf \
+    quay.io/metal3-io/sushy-tools:latest sushy-emulator
+
+  sleep 5
+
+  # Fetch the ID of the system with the name 'bmo-e2e-0'
+  ids=$(curl -s "http://${IP_ADDRESS}:${SUSHY_PORT}/redfish/v1/Systems" | jq -r '.Members[] | .["@odata.id"] | split("/")[-1] | rtrimstr("\"")')
+  # Iterate over each ID and fetch its details
+  for id in ${ids}; do
+    data=$(curl -s "http://${IP_ADDRESS}:${SUSHY_PORT}/redfish/v1/Systems/${id}")
+    name=$(echo "${data}" | jq -r '.Name')
+    if [[ "${name}" == "bmo-e2e-0" ]]; then
+      export BMC_ADDRESS="redfish+http://${IP_ADDRESS}:${SUSHY_PORT}/redfish/v1/Systems/${id}"
+      break
+    fi
+  done
 
-mkdir -p "${IMAGE_FOLDER}"
-pushd "${IMAGE_FOLDER}"
-
-## Setup image server
-# Check if IMAGE_FILE already exists
-if [[ ! -f "${IMAGE_FILE}" ]]; then
-    wget "https://download.cirros-cloud.net/${CIRROS_VERSION}/${IMAGE_FILE}"
 else
-    echo "${IMAGE_FILE} already exists. Skipping download."
+  echo "Invalid e2e emulator specified: ${BMO_E2E_EMULATOR}"
+  exit 1
 fi
 
+# Image server variables
+CIRROS_VERSION="0.6.2"
+IMAGE_FILE="cirros-${CIRROS_VERSION}-x86_64-disk.img"
+export IMAGE_CHECKSUM="c8fc807773e5354afe61636071771906"
+export IMAGE_URL="http://${IP_ADDRESS}/${IMAGE_FILE}"
+IMAGE_DIR="${REPO_ROOT}/test/e2e/images"
+
+## Download and run image server
+mkdir -p "${IMAGE_DIR}"
+pushd "${IMAGE_DIR}"
+wget "https://download.cirros-cloud.net/${CIRROS_VERSION}/${IMAGE_FILE}"
```

Comment:
nit: Wget's progress report is a bit spammy in the logs. I think it would be nice to make it quiet
```suggestion
wget --quiet "https://download.cirros-cloud.net/${CIRROS_VERSION}/${IMAGE_FILE}"
```